fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJobManager.java

```diff
@@ -19,55 +19,160 @@
+        this.idToStatisticsJob.put(statisticsJob.getId(), statisticsJob);
         try {
             Catalog.getCurrentCatalog().getStatisticsJobScheduler().addPendingJob(statisticsJob);
         } catch (IllegalStateException e) {
             LOG.info("The pending statistics job is full. Please submit it again later.");
+            throw new DdlException("The pending statistics job is full, Please submit it again later.");
         }
     }
 
-    // Rule1: The same table cannot have two unfinished statistics jobs
-    // Rule2: The unfinished statistics job could not more then Config.max_statistics_job_num
-    // Rule3: The job for external table is not supported
-    private void checkRestrict(Set<Long> tableIdList) {
-        // TODO
+    /**
+     * The statistical job has the following restrict:
+     * - Rule1: The same table cannot have two unfinished statistics jobs
+     * - Rule2: The unfinished statistics job could not more than Config.max_statistics_job_num
+     * - Rule3: The job for external table is not supported
+     */
+    private void checkRestrict(long dbId, Set<Long> tableIds) throws AnalysisException {
+        Database db = Catalog.getCurrentCatalog().getDbOrAnalysisException(dbId);
+        db.readLock();
+        try {
+            // check table type
+            for (Long tableId : tableIds) {
+                Table table = db.getTableOrAnalysisException(tableId);
+                if (table.getType() != Table.TableType.OLAP) {
+                    ErrorReport.reportAnalysisException(ErrorCode.ERR_NOT_OLAP_TABLE, db.getFullName(), table.getName(), "ANALYZE");
+                }
+            }
+        } finally {
+            db.readUnlock();
+        }
+
+
+        int unfinishedJobs = 0;
+
+        // check table unfinished job
+        for (StatisticsJob statisticsJob : this.idToStatisticsJob.values()) {
+            StatisticsJob.JobState jobState = statisticsJob.getJobState();
+            Set<Long> tblIds = statisticsJob.getTblIds();
+            if (jobState == StatisticsJob.JobState.PENDING
+                    || jobState == StatisticsJob.JobState.SCHEDULING
+                    || jobState == StatisticsJob.JobState.RUNNING) {
+                for (Long tableId : tableIds) {
+                    if (tblIds.contains(tableId)) {
+                        throw new AnalysisException("The table(id=" + tableId + ") have unfinished statistics jobs");
+                    }
+                }
+                unfinishedJobs++;
+            }
+        }
+
+        // check the number of unfinished tasks
+        if (unfinishedJobs > Config.cbo_max_statistics_job_num) {
+            throw new AnalysisException("The unfinished statistics job could not more than cbo_max_statistics_job_num: " +
+                    Config.cbo_max_statistics_job_num);
+        }
     }
 
-    private void checkPermission() {
-        // TODO
+    public void alterStatisticsJobInfo(Long jobId, Long taskId, String errorMsg)  {
```

Comment:
This logic has nothing to do with the statistics manager, but it would be better to put it in statistics jobs.

Reply:
I have removed this method and do same thing elsewhere